src/Utilities/Compiler/Extensions/ITypeSymbolExtensions.cs

```diff
@@ -135,18 +135,71 @@ public static bool ImplementsIDisposable(this ITypeSymbol type, INamedTypeSymbol
         public static bool IsDisposable(this ITypeSymbol type, INamedTypeSymbol iDisposable)
             => type.IsReferenceType && (Equals(type, iDisposable) || type.ImplementsIDisposable(iDisposable));
 
-        public static IEnumerable<AttributeData> GetApplicableAttributes(this INamedTypeSymbol type)
+        public static IEnumerable<AttributeData> GetApplicableAttributes(this INamedTypeSymbol type, INamedTypeSymbol attributeUsageAttribute)
```

Comment:
I don't understand the API change. Earlier API was walking the inheritance chain and returning all applied attributes. The new API seems extremely weird in the sense that it is forcing to pass in a very specific attribute type and then doing some filtering based on it. Can you please give more details on what was the original issue and how is this API change fixing it?

Reply:
Ah, so you are changing the API semantics to now only include directly applied and inherited attributes, not all the attributes on the entire type chain. Can you please add doc comments on the same line?

Reply:
> Ah, so you are changing the API semantics to now only include directly applied and inherited attributes, not all the attributes on the entire type chain

Yes, this would be the definition of "applicable attributes" for a type

Reply:
Now documented